### PR TITLE
FAI-854 - Avoid gradient explosion in LIME perceptron impl

### PR DESCRIPTION
**Pull request by tteofili**
See https://issues.redhat.com/browse/FAI-854

### CODE REVIEW

File: explainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/DummyModelsLimeExplainerTest.java [hunk 1]
```diff
@@ -54,6 +54,7 @@ void testMapOneFeatureToOutputRegression(long seed) throws Exception {
                 .get(Config.INSTANCE.getAsyncTimeout(), Config.INSTANCE.getAsyncTimeUnit());
         for (Saliency saliency : saliencyMap.getSaliencies().values()) {
             assertNotNull(saliency);
+            System.err.println(saliency);
```
Comment (ruivieira) [inline explainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/DummyModelsLimeExplainerTest.java]: @tteofili are these `println` to stay?

**Review by ruivieira — APPROVED** [review]
LGTM

File: explainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/DummyModelsLimeExplainerTest.java (hunk 1 above)
Comment (tteofili) [reply explainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/DummyModelsLimeExplainerTest.java]: of course not, polishing commit underway :smile:

**Comment by tteofili** [comment]
I've incorporated a simple [early stopping](https://en.wikipedia.org/wiki/Early_stopping) mechanism into this PR.
This aims to compensate a lower initial learning rate (1 --> 1e-2) together with an increased number of epochs (50 --> 5000).

**Review by RobGeada — APPROVED** [review]
LGTM, thanks!

**Pull request merged**